server/src/main/java/org/opensearch/identity/ExtensionTokenProcessor.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ *
+ * Modifications Copyright OpenSearch Contributors. See
+ * GitHub history for details.
+ */
+
+package org.opensearch.identity;
+
+import java.security.Principal;
+
+/**
+ * Token processor class to handle token encryption/decryption
+ * This processor is will be instantiated for every extension
+ */
+public class ExtensionTokenProcessor {
```

Comment:
This is very much tied to an extension, do you want to do something more generic which we can use for non-extensions? 

Again, not a blocker for this PR. Trying to understand how we could make it better for rest of the use-cases.